src/components/cc-ssh-key-list/cc-ssh-key-list.types.d.ts

```diff
@@ -0,0 +1,51 @@
+//#region creation form
+interface CreateSshKeyFormState {
+  state: "idle" | "creating";
+  name: FormField;
+  publicKey: FormField;
+}
+//#endregion
+
+//#region key lists
+type KeyDataState = KeyDataStateLoading | KeyDataStateLoadedAndUnlinked | KeyDataStateLoadedAndLinked | KeyDataStateError;
+
+// when exchange with API is occurring = loading SSH keys
+// - is the initial state
+interface KeyDataStateLoading {
+  state: "loading";
+}
+
+// when ready to receive user inputs (personal keys only)
+interface KeyDataStateLoadedAndUnlinked {
+  state: "loaded";
+  isGithubLinked: false;
+  personalKeys: SshKey[];
+}
+
+// when ready to receive user inputs (personal keys & GitHub keys)
+interface KeyDataStateLoadedAndLinked {
+  state: "loaded";
+  isGithubLinked: true;
+  personalKeys: SshKey[];
+  githubKeys: SshKey[];
+}
+
+// when an error has occurred
+interface KeyDataStateError {
+  state: "error";
+}
+//#endregion
+
+//#region common
+interface FormField {
+  value: string;
+  error?: "required";
```

Comment:
The form field is "required". But the error is "empty"

Reply:
Yeah, we need to decided together between empty and required for required text fields.

Reply:
I went for `required` everywhere until we make a decision between `required` and `empty`!